packages/cli/test/dev/utils.js

```diff
@@ -490,30 +508,94 @@ function testFixtureStdio(
       };
       await fn(helperTestPath, port);
     } finally {
-      dev.kill('SIGTERM');
+      await nukeProcessTree(dev.pid);
       await exitResolver;
     }
   };
 }
 
+async function ps(parentPid, pids = {}) {
+  const cmd =
+    process.platform === 'darwin'
+      ? ['pgrep', '-P', parentPid]
+      : ['ps', '-o', 'pid', '--no-headers', '--ppid', parentPid];
+
+  try {
+    const { stdout: buf } = spawnSync(cmd[0], cmd.slice(1), {
+      encoding: 'utf-8',
+    });
+    for (let pid of buf.match(/\d+/g)) {
+      pid = parseInt(pid);
+      const recurse = Object.prototype.hasOwnProperty.call(pids, pid);
+      pids[parentPid].push(pid);
+      pids[pid] = [];
+      if (recurse) {
+        await ps(pid, pids);
+      }
+    }
+  } catch (e) {
+    // squelch
+  }
+  return pids;
+}
+
+async function nukePID(pid, signal = 'SIGTERM', retries = 10) {
+  if (retries === 0) {
+    console.log(`pid ${pid} won't die, giving up`);
+    return;
+  }
+
+  // kill the process
+  try {
+    process.kill(pid, signal);
+  } catch (e) {
+    // process does not exist
+    return;
+  }
+
+  await sleep(250);
+
+  try {
+    // check if killed
+    process.kill(pid, 0);
+  } catch (e) {
+    return;
```

Comment:
**suggestion (non-blocking):** debug log error

Can we debug log this error?

Reply:
Fixed!